packages/notebook/test/modelfactory.spec.ts

```diff
@@ -79,4 +85,72 @@ describe('@jupyterlab/notebook', () => {
       });
     });
   });
+
+  describe('NotebookViewModel', () => {
+    let notebook: NotebookViewModelTest;
+
+    beforeEach(() => {
+      notebook = new NotebookViewModelTest([]);
+    });
+
+    test('should calculate height based on number of lines in source and output (string output)', () => {
+      const outputObj = [
+        {
+          output_type: 'execute_result',
+          data: {
+            'text/plain': '15',
+            'text/html': ['<div>\n', ' <p>15</p>\n', '</div>']
+          },
+          execution_count: 2,
+          metadata: {}
+        }
+      ];
+      const sharedModel = createStandaloneCell({
+        cell_type: 'code',
+        execution_count: 1,
+        outputs: outputObj,
+        source: ['sum([1, 2, 3, 4, 5])'],
+        metadata: {}
+      }) as YCodeCell;
+
+      const model: ICodeCellModel = new CodeCellModel({ sharedModel });
+      expect(model.executionCount).toBe(1);
+      notebook.cells.push({ model } as Cell<ICodeCellModel>);
+
+      const height = notebook.estimateWidgetSize(0);
+      expect(height).toBe(56); // (1 source_line + 1 output_line) * 17 (line_height) + 22 (cell_margin)
+    });
+
+    test('should calculate height based on number of lines in source and output (string[] output)', () => {
+      const outputObj = [
+        {
+          output_type: 'execute_result',
+          data: {
+            'text/plain': ['Output line 1', 'Output line 2', 'Output line 3'],
+            'text/html': []
+          },
+          execution_count: 4,
+          metadata: {}
+        }
+      ];
+      const sharedModel = createStandaloneCell({
+        cell_type: 'code',
+        execution_count: 1,
+        outputs: outputObj,
+        source: ['for i in range(3):\n', '    print(f"Line {i+1}")'],
```

Comment:
Just to align with output:

```suggestion
        source: ['for i in range(3):\n', '    print(f"Output line {i+1}")'],
```